### PR TITLE
WebSerial flasher

### PR DESCRIPTION
**Pull request by puddly**
This is a proof of concept. The `Flasher` class will ideally be exposed as a JavaScript object and then used natively.

image

### CODE REVIEW

**Comment by MattWestb** [comment]
May i a requesting it as one `custom_components` in (Z)HA ?
I like the GUI and user dont need CLI is they is not secure with it and OS and Sup its not so easy to do CLI things with all lock downs of the system.

I also think we need some easy way going back from RCP to NCP or NCP/RCP firmware.

I was testing cooking GSDK 4.2 RCP / EZSP 7.2 for IKEA Marcus and bootloader is working OK both 6.10 and 7.X but the app is not, so very easy testing with your flasher only flashing bootloader with SL Commander and then flashing APP with USF and the SOC is not communicating and using SL commander reading flash and it have allocating NVM = APP running but crashing or comport is not working if the APP.

### Thanks for great work done !!!

File: web/assets/manifests/skyconnect.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
 	"firmwares": [
 		{
 			"name": "Zigbee (EZSP)",
-			"url": "https://raw.githubusercontent.com/NabuCasa/silabs-firmware/main/EmberZNet/beta/NabuCasa_SkyConnect_EZSP_v7.1.3.0_ncp-uart-hw_115200.gbl",
+			"url": "https://raw.githubusercontent.com/NabuCasa/silabs-firmware/main/EmberZNet/beta/NabuCasa_SkyConnect_EZSP_v7.1.4.0_ncp-uart-hw_115200.gbl",
 			"type": "ncp-uart-hw",
 			"version": "7.1.3.0"
```
Comment (MattWestb) [inline web/assets/manifests/skyconnect.json]: "7.1.4.0"  ??

Comment (puddly) [reply web/assets/manifests/skyconnect.json]: Oops. Thank you!

**Comment by puddly** [comment]
@MattWestb The web flasher has been moved to https://github.com/NabuCasa/sl-web-tools.

You can try it live (https://skyconnect.home-assistant.io/, scroll to the bottom), or build it yourself locally:

```bash
git clone https://github.com/NabuCasa/sl-web-tools
cd sl-web-tools
npm install .
./script/develop
```

**Comment by MattWestb** [comment]
Thanks i trying it later then its starts being little late :-))

**Comment by MattWestb** [comment]
SC01

**Pull request closed without merging**